File: scripts/fix_timeline.py

import os

file_path = r"c:\Users\amiri\FavoritePart\src\components\PlayerTimeline.tsx"

with open(file_path, 'r', encoding='utf-8') as f:
    lines = f.readlines()

new_lines = []
skip = False

# We know the duplicate block starts after the NEW block ends.
# The NEW block ends with `            })()}` (plus garbage `ips...`) at line 1388 (1-indexed) -> index 1387.
# The duplicate block ends at line 1753 (1-indexed) -> index 1752.

# Let's inspect line 1387
print(f"Line 1387: {lines[1387]}")
# Expected: "            })()}ips (Bi-directional)\n"

# We will correct line 1387
lines[1387] = "            })()}\n"

# We want to KEEP lines 0 to 1387.
new_lines.extend(lines[:1388])

# We want to SKIP lines 1388 to 1752 (inclusive of 1752 which is `            })()}`)
# Wait, line 1753 (index 1752) is `            })()}`.
# This logic is tricky. 
# The NEW block is 1067 to 1388.
# The OLD block should be removed.
# The OLD block starts at 1389 (index 1388).
# The OLD block ends at 1753 (index 1752).
# We want to keep 1754 (index 1753) onwards.

print(f"Line 1753 (should be skipped): {lines[1752]}")
print(f"Line 1754 (should be kept): {lines[1753]}")

new_lines.extend(lines[1753:])

with open(file_path, 'w', encoding='utf-8') as f:
    f.writelines(new_lines)

print("Fixed PlayerTimeline.tsx")

File: scripts/fix_timeline.js
const fs = require('fs');
const path = require('path');

const filePath = path.join(process.cwd(), 'src/components/PlayerTimeline.tsx');

try {
    const content = fs.readFileSync(filePath, 'utf8');
    const lines = content.split('\n');

    // Index 1387 is Line 1388
    console.log(`Line 1388: ${lines[1387]}`);

    // Fix Line 1388
    lines[1387] = lines[1387].replace('})()}ips (Bi-directional)', '})()}');
    console.log(`Fixed Line 1388: ${lines[1387]}`);

    // Keep 0 to 1387 (inclusive) -> lines.slice(0, 1388)
    const part1 = lines.slice(0, 1388);

    // Skip 1388 to 1752 (inclusive). 
    // Line 1753 (index 1752) is the duplicate `})()}`.
    // We want to KEEP from Line 1754 (index 1753) onwards.
    const part2 = lines.slice(1753);

    console.log(`Line 1754 (kept): ${part2[0]}`);

    const newContent = [...part1, ...part2].join('\n');
    fs.writeFileSync(filePath, newContent, 'utf8');

    console.log('Fixed PlayerTimeline.tsx successfully');

} catch (err) {
    console.error('Error:', err);
}
